File: 110.BalancedBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    boolean balanced = true;

    int check(TreeNode root) {
        
        if (root == null){
            return 0;
        }

        int lh = check(root.left);
        int rh = check(root.right);
        if (Math.abs(lh - rh) > 1){
            balanced = false;
        }

        int height = Math.max(lh, rh) + 1;
        return height;
    }

    public boolean isBalanced(TreeNode root) {
        check(root);
        return balanced;
        
    }
}

File: 110.BalancedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def __init__(self):
        self.balanced = True


    def check(self, root):
        if root == None:
            return 0
        
        lh = self.check(root.left)
        rh = self.check(root.right)

        if abs(lh - rh) > 1:
            self.balanced = False

        return max(lh, rh) + 1

    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.check(root)
        return self.balanced
